### PR TITLE
Get the game running in Ren'Py 7.6

### PR DESCRIPTION
**Pull request by einstein95**
Also adding more stuff to .gitignore so that game stuff doesn't get flagged by git.

Only thing now is to fix the recursion error when renpy.quit() is called

### CODE REVIEW

**Comment by einstein95** [comment]
@Blizzardsev that's fair. The recursion error seems to be caused by zz_config.rpy setting `config.quit_action` to be `quit_input_check`, and then that function jumping to `quit` which then calls `renpy.quit()` which calls that function...

**Comment by Blizzardsev** [comment]
Seeing as how that's current functionality, then my guess would be a change in the behaviour of `renpy.quit()` between versions: a good place to start would be going through the changelogs for Ren'Py post-7 and seeing where/if the behaviour of that function is modified, and how.

If `renpy.quit()` now has a hard link to `config.quit_action` instead of just quitting the game, then we'd need to either see if the changes add a parameter we can use to stop that, use a replacement function or try dummying out `config.quit_action` just before we call it.

**Comment by einstein95** [comment]
I was incorrect in debugging the error, it was caused by the label being called `quit`, which `renpy.quit()` jumps to if it's defined (introduced by https://github.com/renpy/renpy/commit/e281635b471e34c9f6a62566877c56bd9e024c5d). I've renamed the label to `save_quit` (as it saves, then quits) and it fixes the error.

The TODO introduced in this PR does not seem to affect the game, as it seems to only affect the transition when bringing up the menu of topics to talk about. Further regression testing may be needed.

**Comment by Blizzardsev** [comment]
Good stuff - always nice for the solution to be quick and easy, well done.

See if you can get a handle on that transition if you can: the categorized menu is pretty front and centre visually being used for the main game menu, so we would want to preserve that behaviour if possible: might be as providing the `at` statement at the point of showing the screen.

**Comment by einstein95** [comment]
After comparing the behavior in 7.6 and the current RenPy 6 version, it doesn't change anything. From what I can see, it was added in at some point so that the list of topics would slide in from the right, however I think this functionality broke at some point without people noticing or never worked.

File: game/screens.rpy [hunk 1]
```diff
@@ -127,7 +127,8 @@ style categorized_menu_button_text_italic is categorized_menu_button_text:
     italic True
 
 screen categorized_menu(menu_items, category_pane_space, option_list_space, category_length):
-    at categorized_menu_slide_in_right
+    # TODO: u'at' is not a keyword argument or valid child of the screen statement.
+    # at categorized_menu_slide_in_right
```
Comment (Blizzardsev) [inline game/screens.rpy:131]: Can we remove this please?

**Review by Blizzardsev — CHANGES_REQUESTED** [review]
Minor changes. My main concern here is going to be the amount of changes to packages as a result of the move: particularly those related to making requests.

Have you tested the weather-related setup to ensure that still works as expected?

File: game/screens.rpy (hunk 1 above)
Comment (einstein95) [reply game/screens.rpy]: Done

**Comment by einstein95** [comment]
> Minor changes. My main concern here is going to be the amount of changes to packages as a result of the move: particularly those related to making requests.
> 
> Have you tested the weather-related setup to ensure that still works as expected?

Yes, I've had the weather change perfectly fine.

Since Renpy 7 is still python 7, the packages haven't needed updating, I've only done so for these packages so that they'll still continue to work for the near future and to smooth over any further update to Renpy 8, as Renpy 7 will only be supported for a year.

**Review by Blizzardsev — APPROVED** [review]

**Comment by Blizzardsev** [comment]
Cool. I'd like to run through the engine change in more detail personally, but until then I'll merge it in.

**Pull request merged**